### PR TITLE
Avoid all-zeor attnetion mask used in testing

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

The method `random_attention_mask` used in testing makes sure the last token is non-zero. However, this property will be changed if a causal mask is applied.

This causes some issues in CI, see issue reported 

https://github.com/pytorch/pytorch/issues/110213

In general, a sequence with all zero as attention mask is bad. Let's avoid testing with such case.

(However, we probably need to do some processing in the modeling code - if torch decide this is undefined behavior and won't make change to have previous behavior).

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Thaks @ydshieh!
